tests/test_errors.py

```diff
@@ -170,11 +170,88 @@ def get_model(compile, meta, compile_kwargs):
 
     est = KerasRegressor(
         model=get_model,
-        loss="categorical_crossentropy",
+        loss="sparse_categorical_crossentropy",
         compile=compile,
         optimizer=None,
     )
     with pytest.raises(
         ValueError, match="Could not interpret optimizer identifier"  # Keras error
     ):
-        est.fit([[1]], [1])
+        est.fit(np.array([[0], [1]]), np.array([0, 1]))
+
+
+def test_target_dtype_changes_incremental_fit():
+    X = np.array([[1, 2], [2, 3]])
+    y = np.array([1, 3])
+
+    est = KerasClassifier(model=dynamic_classifier, hidden_layer_sizes=(100,))
+    est.fit(X, y)
+    est.partial_fit(X, y.astype(np.uint8))
+    with pytest.raises(
+        ValueError, match=" and casting from ",
+    ):
+        est.partial_fit(X, y.astype(np.float64))
+
+
+def test_target_dims_changes_incremental_fit():
+    X = np.array([[1, 2], [2, 3]])
+    y = np.array([1, 3])
+
+    est = KerasClassifier(model=dynamic_classifier, hidden_layer_sizes=(100,))
+    est.fit(X, y)
+    y_new = y.reshape(-1, 1)
+    with pytest.raises(
+        ValueError, match="`y` has 2 dimensions, but this ",
+    ):
+        est.partial_fit(X, y_new)
+
+
+def test_target_shape_changes_incremental_fit_clf():
+    X = np.array([[1, 2], [2, 3]])
+    y = np.array([1, 3]).reshape(-1, 1)
+
+    est = KerasClassifier(model=dynamic_classifier, hidden_layer_sizes=(100,))
+    est.fit(X, y)
+    with pytest.raises(
+        ValueError, match="The number of features ",  # raised by transformers
```

Comment:
Is this an error on the number of features or the number of labels/targets?

Reply:
It's because of `y`, but remember that transformers operate on `X`. I see how this could be confusing, but the only way around it is to catch the error and reword it.

Reply:
We could change the part of the message we look for to make the test less confusing: `match="The fitted data had "`

Reply:
I see. I think the traceback will make this information clear. This thread can be resolved.